pkg/operator/validation.go

```diff
@@ -0,0 +1,178 @@
+		errs = append(errs, fmt.Errorf("prestoClientCertFile and prestoClientKeyFile must both be specified or neither specified"))
+	}
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidHiveConfig ensure all Hive* fields are valid if provided.
+func IsValidHiveConfig(cfg *Config) error {
+	errs := []error{}
+
+	errs = append(errs, isValidHostPort(cfg.HiveHost, "hiveHost"))
+
+	if !cfg.HiveUseTLS {
+		if cfg.HiveUseClientCertAuth {
+			errs = append(errs, fmt.Errorf("hiveUseClientCertAuth cannot be set to true if hiveUseTLS is false"))
+		}
+
+		if len(cfg.HiveCAFile) > 0 {
+			errs = append(errs, fmt.Errorf("hiveCAFile cannot be set if hiveUseTLS is false"))
+		}
+	}
+
+	if len(cfg.HiveCAFile) > 0 {
+		if _, err := os.Stat(cfg.HiveCAFile); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	if (len(cfg.HiveClientCertFile) > 0 && len(cfg.HiveClientKeyFile) == 0) ||
+		(len(cfg.HiveClientKeyFile) > 0 && len(cfg.HiveClientCertFile) == 0) {
+		errs = append(errs, fmt.Errorf("hiveClientCertFile and hiveClientKeyFile must both be specified or neither specified"))
+	}
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidKubeConfig ensures the kube config is set to a valid file if provided.
+func IsValidKubeConfig(kubeconfig string) error {
+	if len(kubeconfig) > 0 {
+		if _, err := os.Stat(kubeconfig); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// IsValidPrometheusConfig ensures prometheus configuration is valid.
+func IsValidPrometheusConfig(cfg *Config) error {
+	errs := []error{}
+	if cfg.PrometheusConfig.CAFile != "" {
+		if _, err := os.Stat(cfg.PrometheusConfig.CAFile); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	// PrometheusDataSourceMaxBackfillImportDuration overrides PrometheusDataSourceGlobalImportFromTime
+	// don't set both.
+	if cfg.PrometheusDataSourceGlobalImportFromTime != nil && cfg.PrometheusDataSourceMaxBackfillImportDuration > 0 {
+		errs = append(errs, fmt.Errorf("prometheusDataSourceGlobalImportFromTime and prometheusDataSourceMaxBackfillImportDuration cannot both be set"))
+	}
+
+	if len(errs) > 0 {
+		return errors.NewAggregate(errs)
+	}
+	return nil
+}
+
+// IsValidTLSConfig ensures the TLS config is valid.
+func isValidTLSConfig(cfg *TLSConfig) error {
+	if cfg.UseTLS {
+		if cfg.TLSCert == "" {
+			return fmt.Errorf("must set TLS certificate if TLS is enabled")
+		}
+		if cfg.TLSKey == "" {
+			return fmt.Errorf("must set TLS private key if TLS is enabled")
+		}
+	}
+	return nil
+}
+
+// isValidHostPort attempts to split a non empty hp into host and port, returning any errors found.
+// TODO this is only validating non-empty strings.  We may want to check for empty strings an report errors.
+// TODO this requires a port to be specified, is that one of our requirements?
+func isValidHostPort(hp string, name string) error {
+	if len(hp) > 0 {
+		if _, _, err := net.SplitHostPort(hp); err != nil {
```

Comment:
this will fail if a port is not given - can someone confirm that that is a correct behavior?

Reply:
This gets called on api, metrics, pprof, presto, hive
 
I think it's valid to make the admin config the port for our API.
Metrics = Prom, also has a default.
pprof seems like it has a default port as well, so maybe okay to not spec.
Presto has a default port, might be ok to not spec.
Hive has a default thrift service port as well, so maybe that might not need being requeired to specify either.

Reply:
Ok, I will remove the calls/tests for apis that have default ports.